crates/bevy_gltf/src/loader.rs

```diff
@@ -77,34 +83,249 @@ impl AssetLoader for GltfLoader {
-            supported_compressed_formats,
+        match format {
+            (DataType::F32, Dimensions::Scalar) => acc.with_no_norm(VertexAttributeIter::F32),
+            (DataType::U32, Dimensions::Scalar) => acc.with_no_norm(VertexAttributeIter::U32),
+            (DataType::F32, Dimensions::Vec2) => acc.with_no_norm(VertexAttributeIter::F32x2),
+            (DataType::U32, Dimensions::Vec2) => acc.with_no_norm(VertexAttributeIter::U32x2),
+            (DataType::F32, Dimensions::Vec3) => acc.with_no_norm(VertexAttributeIter::F32x3),
+            (DataType::U32, Dimensions::Vec3) => acc.with_no_norm(VertexAttributeIter::U32x3),
+            (DataType::F32, Dimensions::Vec4) => acc.with_no_norm(VertexAttributeIter::F32x4),
+            (DataType::U32, Dimensions::Vec4) => acc.with_no_norm(VertexAttributeIter::U32x4),
+            (DataType::I16, Dimensions::Vec2) => acc.with_norm(VertexAttributeIter::S16x2),
+            (DataType::U16, Dimensions::Vec2) => acc.with_norm(VertexAttributeIter::U16x2),
+            (DataType::I16, Dimensions::Vec4) => acc.with_norm(VertexAttributeIter::S16x4),
+            (DataType::U16, Dimensions::Vec4) => acc.with_norm(VertexAttributeIter::U16x4),
+            (DataType::I8, Dimensions::Vec2) => acc.with_norm(VertexAttributeIter::S8x2),
+            (DataType::U8, Dimensions::Vec2) => acc.with_norm(VertexAttributeIter::U8x2),
+            (DataType::I8, Dimensions::Vec4) => acc.with_norm(VertexAttributeIter::S8x4),
+            (DataType::U8, Dimensions::Vec4) => acc.with_norm(VertexAttributeIter::U8x4),
+            (DataType::U16, Dimensions::Vec3) => acc.with_norm(VertexAttributeIter::U16x3),
+            (DataType::U8, Dimensions::Vec3) => acc.with_norm(VertexAttributeIter::U8x3),
+            _ => Err(AccessFailed::UnsupportedFormat),
+        }
+    }
+
+    /// Materializes values for any supported format of vertex attribute
+    fn into_any_values(self) -> Result<VertexAttributeValues, AccessFailed> {
+        match self {
+            VertexAttributeIter::F32(it) => Ok(VertexAttributeValues::Float32(it.collect())),
+            VertexAttributeIter::U32(it) => Ok(VertexAttributeValues::Uint32(it.collect())),
+            VertexAttributeIter::F32x2(it) => Ok(VertexAttributeValues::Float32x2(it.collect())),
+            VertexAttributeIter::U32x2(it) => Ok(VertexAttributeValues::Uint32x2(it.collect())),
+            VertexAttributeIter::F32x3(it) => Ok(VertexAttributeValues::Float32x3(it.collect())),
+            VertexAttributeIter::U32x3(it) => Ok(VertexAttributeValues::Uint32x3(it.collect())),
+            VertexAttributeIter::F32x4(it) => Ok(VertexAttributeValues::Float32x4(it.collect())),
+            VertexAttributeIter::U32x4(it) => Ok(VertexAttributeValues::Uint32x4(it.collect())),
+            VertexAttributeIter::S16x2(it, n) => Ok(n.apply_either(
+                it.collect(),
+                VertexAttributeValues::Snorm16x2,
+                VertexAttributeValues::Sint16x2,
```

Comment:
This might be controversial, but I'd like a `use VertexAttributeValues as Attrs` at the start of the function scope, so that calls to `apply_either` are collapsed into a single line.